cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/ACLineSegmentConversion.java

```diff
@@ -41,27 +43,51 @@ public boolean valid() {
 
     @Override
     public void convert() {
+        convertLine();
+    }
+
+    @Override
+    public void convertAtBoundary() {
         if (isBoundary(1)) {
             convertLineAtBoundary(1);
         } else if (isBoundary(2)) {
             convertLineAtBoundary(2);
         } else {
-            convertLine();
+            throw new ConversionException("Boundary must be at one end of the line");
         }
     }
 
-    private void convertLineAtBoundary(int boundarySide) {
-        // If we have created buses and substations for boundary nodes,
-        // convert as a regular line
-        if (context.config().convertBoundary()) {
-            convertLine();
+    @Override
+    public BoundaryLine asBoundaryLine(String boundaryNode) {
+        BoundaryLine boundaryLine = super.createBoundaryLine(boundaryNode);
+        double r = p.asDouble("r");
+        double x = p.asDouble("x");
+        double g = p.asDouble("gch", 0);
+        double b = p.asDouble("bch", 0);
+        boundaryLine.setParameters(r, x, g / 2, b / 2, g / 2, b / 2);
+        return boundaryLine;
+    }
+
+    public static void convertBoundaryLines(Context context, String boundaryNode, BoundaryLine boundaryLine1, BoundaryLine boundaryLine2) {
+        Line mline;
+        if (context.config().mergeBoundariesUsingTieLines()) {
+            mline = createTieLine(context, boundaryNode, boundaryLine1, boundaryLine2);
         } else {
-            double r = p.asDouble("r");
-            double x = p.asDouble("x");
-            double gch = p.asDouble("gch", 0.0);
-            double bch = p.asDouble("bch");
+            mline = createLine(context, boundaryLine1, boundaryLine2);
+        }
 
-            convertToDanglingLine(boundarySide, r, x, gch, bch);
+        mline.addAlias(boundaryLine1.getModelTerminalId(), Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + CgmesNames.TERMINAL + 1);
+        mline.addAlias(boundaryLine1.getBoundaryTerminalId(), Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + "HALF1." + CgmesNames.TERMINAL + "_BOUNDARY");
+        mline.addAlias(boundaryLine2.getModelTerminalId(), Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + CgmesNames.TERMINAL + 2);
+        mline.addAlias(boundaryLine2.getBoundaryTerminalId(), Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + "HALF2." + CgmesNames.TERMINAL + "_BOUNDARY");
+
+        context.convertedTerminal(boundaryLine1.getModelTerminalId(), mline.getTerminal1(), 1, boundaryLine1.getModelPowerFlow());
+        context.convertedTerminal(boundaryLine2.getModelTerminalId(), mline.getTerminal2(), 2, boundaryLine2.getModelPowerFlow());
+
+        if (mline instanceof TieLine) {
```

Comment:
Same remark: if it is always a tie line, we don't need this condition anymore

Reply:
I will remove it in a separated PR after merging this one.